wayland-client/src/rust_imp/mod.rs

```diff
@@ -183,3 +196,39 @@ pub(crate) fn default_dispatcher() -> Arc<Mutex<dyn Dispatcher + Send>> {
 
     Arc::new(Mutex::new(DefaultDisp))
 }
+
+/// Handler for printing events when `WAYLAND_DEBUG` is `true`.
+fn print_event(
+    interface: &str,
+    id: u32,
+    msg_name: &str,
+    request_name: Option<&str>,
+    args: &[Argument],
+) {
```

Comment:
This function can be moved to `wayland-commons`, so that it can be re-used in `wayland-server` as well.

However, I don't really like the way `msg_name` is overloaded like that to mean something different and how the direction of the message (event or request) is inferred from that. I'd rather have an explicit `bool` argument specifying if the message is sent or received, and an other for the `[ZOMBIE]` annotation.